### PR TITLE
[issue:1] Twitter-IFTTT driven microservice demo

### PR DESCRIPTION
**Pull request by nitind**
(c) Copyright IBM Corp. 2016

### CODE REVIEW

**Comment by nitind** [comment]
Demo and readme describing Twitter-IFTTT setup.

File: microservice_demo/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM jupyter/datascience-notebook
```
Comment (parente) [inline microservice_demo/Dockerfile:3]: Add copyright header.

File: microservice_demo/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+The [Kernel Gateway](https://github.com/jupyter-incubator/kernel_gateway) is a
+[JupyterApp](https://github.com/jupyter/jupyter_core/blob/master/jupyter_core/application.py) that
+implements different APIs and protocols for accessing Jupyter kernels.  We can use the Jupyter kernel gateway to
+[process requests](https://github.com/jupyter-incubator/kernel_gateway#processing-requests) and transform an
+annotated notebook into a HTTP API using the Jupyter kernel gateway.
+ 
+This example shows how one can use the Jupyter kernel gateway to deploy an notebook as a
+microservice at the heart of a Twitter bot, using a combination of [Alchemy API](http://www.alchemyapi.com/products/alchemylanguage/entity-extraction) with IFTTT's [Twitter](https://ifttt.com/twitter) and [Maker](https://ifttt.com/maker) channels, to respond to mentions with a search URL for relevant Meetups. You will need developer keys for [Alchemy](http://www.alchemyapi.com/api/register.html) and [Maker](https://ifttt.com/maker) to proceed, as well as an arbitrary Event Name for use with Maker.
+
+ ### Build the docker container
```
Comment (parente) [inline microservice_demo/README.md]: Remove space else it doesn't render as a header. Same on the next one.

File: microservice_demo/README.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ ```
+ make build
+ ```
+
+ ### Run the docker container for the service
+>After having set the ALCHEMY_KEY, MAKER_KEY, and MAKER_EVENT_NAME environment variables accordingly. These variables are passed into the docker container, read and used by the notebook, and must still be set if running the Jupyter kernel gateway outside of a container.
```
Comment (parente) [inline microservice_demo/README.md]: Remove >.

File: microservice_demo/README.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+ ```
+ make run
```
Comment (parente) [inline microservice_demo/README.md]: Show the env vars getting set in the code block. You can do it right on the run command like

```
make run ALCHEMY_KEY=your_key
```

and so on for the rest.

File: microservice_demo/meetup_service.ipynb [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# GET /_city\n",
+    "'''\n",
+    "Check an arbitrary string against the Alchemy entity extraction, using the set API key\n",
```
Comment (parente) [inline microservice_demo/meetup_service.ipynb]: These strings (_Check an arbitrary string ..._) in the code cells are evaled as strings every time the code cell runs. Move the comments to markdown above the cell.

File: microservice_demo/README.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+The [Kernel Gateway](https://github.com/jupyter-incubator/kernel_gateway) is a
+[JupyterApp](https://github.com/jupyter/jupyter_core/blob/master/jupyter_core/application.py) that
+implements different APIs and protocols for accessing Jupyter kernels.  We can use the Jupyter kernel gateway to
+[process requests](https://github.com/jupyter-incubator/kernel_gateway#processing-requests) and transform an
+annotated notebook into a HTTP API using the Jupyter kernel gateway.
+ 
+This example shows how one can use the Jupyter kernel gateway to deploy an notebook as a
+microservice at the heart of a Twitter bot, using a combination of [Alchemy API](http://www.alchemyapi.com/products/alchemylanguage/entity-extraction) with IFTTT's [Twitter](https://ifttt.com/twitter) and [Maker](https://ifttt.com/maker) channels, to respond to mentions with a search URL for relevant Meetups. You will need developer keys for [Alchemy](http://www.alchemyapi.com/api/register.html) and [Maker](https://ifttt.com/maker) to proceed, as well as an arbitrary Event Name for use with Maker.
+
```
Comment (parente) [inline microservice_demo/README.md:11]: Include the name of the running bot and the API endpoint so people can try the live thing without deploying it themselves.

**Pull request merged**